### ISSUE TITLE
External storage (s3) archive job fails silently (api/website)

### ISSUE THREAD

**Issue by TristanWasTaken** [go-gitea/gitea#30443]
### Description

If the s3 storage endpoint is set to a domain which doesn't resolve for whatever reason, the archive download on the user's side silently fails.
While the job throws some errors in the console, the api response only reflects if the archive job is completed and doesn't include a scenario if the job failed, which means it stays as false even after the job failed and isn't being worked on anymore.
`HTTP POST https://git.example.com/tachiyomi/tachiyomi/archive/v0.15.3.zip`
res: `{"complete":false}`
The website/frontend then continues to request the status of the job again and again until the user gets bored and closes the tab.

tldr; archive job fail is not handled by website or api (silent fail)
The 1. api response should include a scenario where the job failed and 2. the frontend should show an error if it did and 3. stop making requests.

### Gitea Version

1.21.3

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

https://gist.github.com/TristanWasTaken/369af8ffdcd7d2e93c507f5fe2c85602

### Operating System

Docker

### How are you running Gitea?

docker `gitea/gitea:1.21.3`
docker `minio/minio`
both run in the same bridge network, though gitea has to contact storage.*example.com* because you can only set one hostname which has to be the public hostname when `SERVE_DIRECT` is used

### Database

None

**Comment by lunny** [comment]
Please remove all `STORAGE_TYPE = minio` from sections except `[storage]` and try again.
